pocld/daemon.cc

```diff
@@ -267,7 +273,8 @@ static std::string find_default_ip_address() {
   return std::string(listen_addr ? listen_addr : "127.0.0.1");
 }
 
-int PoclDaemon::launch(std::string ListenAddress, struct ServerPorts &Ports) {
+int PoclDaemon::launch(std::string ListenAddress, struct ServerPorts &Ports,
+                       const bool UseVsock) {
```

Comment:
For strings and structs in general I can buy it to reduce copying, but for plain old small datatypes that should fit to registers it's difficult to see why.

Reply:
This boolean here is not a reference though, just const, making the compiler yell at you if you accidentally try to change it within the function which would likely have "fun" consequences. That aside, all of this would probably be better passed as an argument to the `PoclDaemon` constructor (which currently just uses the default constructor).

Reply:
Yes, in general I'd agree on the 'const', but my gut says that 99% of args are not typically (meant to be) modified within the function and still they do not mark them const (just clutters the arg list).

Reply:
so change args to bool UseVsock ?